### PR TITLE
WIP: Update tests for VTU beta-31

### PR DESCRIPTION
**Pull request by lmiller1990**
## Description

- Upgrade `@vue/test-utils` to latest version (beta-31)
- Refactor some tests
- Try not to use `setProps`, `setData` (potential depreciation in the future)

## Motivation and Context

- always good be on the latest version
- by doing this, I can also learn the suite better and help you upgrade to a Vue 3 compatible VTU in the future as well

## How Has This Been Tested?

- make sure the tests continue to pass, snapshots are not failing

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [ x The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/vuetify/src/components/VAlert/__tests__/VAlert.spec.ts [hunk 1]
```diff
@@ -60,7 +63,7 @@ describe('VAlert.ts', () => {
     })
 
     const icon = wrapper.find('.v-alert__dismissible')
-    const input = jest.fn(show => wrapper.setProps({ show }))
```
Comment (lmiller1990) [inline packages/vuetify/src/components/VAlert/__tests__/VAlert.spec.ts]: looks like this wasn't doing anything, removing it made no difference

File: packages/vuetify/src/components/VAlert/__tests__/VAlert.spec.ts [hunk 2]
```diff
@@ -89,26 +92,22 @@ describe('VAlert.ts', () => {
...
+  })
 
-    wrapper.setProps({ type: 'success' })
-    await wrapper.vm.$nextTick()
-    expect(wrapper.classes('success')).toBe(true)
-
-    wrapper.setProps({ type: 'warning' })
-    await wrapper.vm.$nextTick()
-    expect(wrapper.classes('warning')).toBe(true)
+  ;['success', 'error', 'warning', 'info'].forEach(type => {
```
Comment (lmiller1990) [inline packages/vuetify/src/components/VAlert/__tests__/VAlert.spec.ts]: can avoid using `setProps` and just test all four colors using a simple for loop - no `await nextTick` is needed here, seems legit

File: packages/vuetify/src/components/VAlert/__tests__/VAlert.spec.ts [hunk 3]
```diff
@@ -89,26 +92,22 @@ describe('VAlert.ts', () => {
     expect(wrapper.contains('.v-icon')).toBe(false)
   })
 
-  // TODO: this fails without sync, nextTick doesn't help
-  // https://github.com/vuejs/vue-test-utils/issues/1130
-  it.skip('should display contextual colors by type', async () => {
+  it('should display contextual colors by type', () => {
```
Comment (KaelWD) [inline packages/vuetify/src/components/VAlert/__tests__/VAlert.spec.ts]: Should be merged with the one below

File: packages/vuetify/src/components/VAlert/__tests__/VAlert.spec.ts [hunk 4]
```diff
@@ -36,9 +34,14 @@ describe('VAlert.ts', () => {
 
     expect(wrapper.element.style.display).toBe('')
     expect(wrapper.html()).toMatchSnapshot()
+  })
 
-    wrapper.setProps({ value: false })
-    await wrapper.vm.$nextTick()
+  it('should sets display to none when value is false', async () => {
```
Comment (KaelWD) [inline packages/vuetify/src/components/VAlert/__tests__/VAlert.spec.ts]:
```suggestion
  it('should be hidden when value is false', async () => {
```

File: packages/vuetify/src/components/VAlert/__tests__/VAlert.spec.ts (hunk 1 above)
Comment (KaelWD) [reply packages/vuetify/src/components/VAlert/__tests__/VAlert.spec.ts]: I think that was supposed to be like v-model from before it stored its own state.

**Comment by lmiller1990** [comment]
We will do v1 for VTU soon. Let's wait for that. I will use this branch and rebase soon.

**Comment by johnleider** [comment]
This Pull Request is being closed due to inactivity.

If you have any additional questions, please reach out to us in our [Discord community](https://community.vuetifyjs.com).

**Pull request closed without merging**